drivers/serial/uart_nrfx_uarte.c

```diff
@@ -251,6 +307,453 @@ static int uarte_nrfx_config_get(struct device *dev, struct uart_config *cfg)
 	return 0;
 }
 
+
+static int uarte_nrfx_err_check(struct device *dev)
+{
+	u32_t error = 0U;
+	NRF_UARTE_Type *uarte = get_uarte_instance(dev);
+
+	if (nrf_uarte_event_check(uarte, NRF_UARTE_EVENT_ERROR)) {
+		/* register bitfields maps to the defines in uart.h */
+		error = nrf_uarte_errorsrc_get_and_clear(uarte);
+	}
+
+	return error;
+}
+
+#ifdef CONFIG_UART_ASYNC_API
+
+static int uarte_nrfx_tx(struct device *dev, const u8_t *buf, size_t len,
+			 u32_t timeout)
+{
+	struct uarte_nrfx_data *data = get_dev_data(dev);
+	const struct uarte_nrfx_config *cfg = get_dev_config(dev);
+
+	if (data->async->tx) {
+		return -EBUSY;
+	}
+	data->async->tx = buf;
+	data->async->tx_len = len;
+	data->async->tx_cur_idx = 0;
+	nrf_uarte_tx_buffer_set(cfg->uarte_regs, buf, len);
+	nrf_uarte_event_clear(cfg->uarte_regs, NRF_UARTE_EVENT_TXSTARTED);
+	nrf_uarte_task_trigger(cfg->uarte_regs, NRF_UARTE_TASK_STARTTX);
+	if (cfg->uarte_regs->CONFIG & UARTE_CONFIG_HWFC_Msk) {
```

Comment:
Why not `if (cfg->flow_ctrl == UART_CFG_FLOW_CTRL_RTS_CTS)`?
We avoid operating directly on the hardware registers in any layers upper than HAL. If the HAL is missing some required functionality, it needs to be extended.